hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/codec/ScheduleServiceStateTranslator.java

```diff
@@ -0,0 +1,249 @@
+        if (scheduleAccount != null) {
+            scheduleBuilder.schedulerAccount(AccountID.newBuilder()
+                    .accountNum(scheduleAccount.num())
+                    .realmNum(scheduleAccount.realm())
+                    .shardNum(scheduleAccount.shard())
+                    .build());
+        }
+
+        final var payerAccount = scheduleVirtualValue.payer();
+        if (payerAccount != null) {
+            scheduleBuilder.payerAccount(AccountID.newBuilder()
+                    .accountNum(payerAccount.num())
+                    .realmNum(payerAccount.realm())
+                    .shardNum(payerAccount.shard())
+                    .build());
+        }
+
+        final var adminKey = scheduleVirtualValue.adminKey();
+        if (adminKey.isPresent()) {
+            scheduleBuilder.adminKey(PbjConverter.asPbjKey(adminKey.get()));
+        }
+
+        final var validStart = scheduleVirtualValue.schedulingTXValidStart();
+        if (validStart != null) {
+            scheduleBuilder.scheduleValidStart(Timestamp.newBuilder()
+                    .seconds(validStart.getSeconds())
+                    .nanos(validStart.getNanos())
+                    .build());
+        }
+
+        final var expirationTime = scheduleVirtualValue.expirationTimeProvided();
+        if (expirationTime != null) {
+            scheduleBuilder.expirationTimeProvided(Timestamp.newBuilder()
+                    .seconds(expirationTime.getSeconds())
+                    .nanos(expirationTime.getNanos())
+                    .build());
+        }
+
+        final var calculatedExpirationTime = scheduleVirtualValue.calculatedExpirationTime();
+        if (calculatedExpirationTime != null) {
+            scheduleBuilder.calculatedExpirationTime(Timestamp.newBuilder()
+                    .seconds(calculatedExpirationTime.getSeconds())
+                    .nanos(calculatedExpirationTime.getNanos())
+                    .build());
+        }
+
+        final var resolutionTime = scheduleVirtualValue.getResolutionTime();
+        if (resolutionTime != null) {
+            scheduleBuilder.resolutionTime(Timestamp.newBuilder()
+                    .seconds(resolutionTime.getSeconds())
+                    .nanos(resolutionTime.getNanos())
+                    .build());
+        }
+
+        final var scheduledTxn = PbjConverter.toPbj(scheduleVirtualValue.scheduledTxn());
+        if (scheduledTxn.isPresent()) {
+            scheduleBuilder.scheduledTransaction(scheduledTxn.get());
+        }
+
+        final var body = scheduleVirtualValue.bodyBytes();
+        if (body.length > 0) {
+            scheduleBuilder.originalCreateTransaction(TransactionBody.PROTOBUF.parse(BufferedData.wrap(body)));
+        }
+
+        var keys = scheduleVirtualValue.signatories();
+        if (keys != null && !keys.isEmpty()) {
+            final List<Key> keyList = new ArrayList<>();
+            for (byte[] key : keys) {
```

Comment:
Nit: missing a couple `final`s

`final var keys = `
and
`       for (final byte[] key : keys) {`

A few methods throughout this class are missing `final` on the input params

Reply:
fixed.